pkg/pub_integration/lib/src/fake_test_context_provider.dart

```diff
@@ -65,11 +67,88 @@ class TestContextProvider {
     required String email,
     List<String>? scopes,
   }) async {
-    return await createFakeTestUser(
+    return await _createFakeTestUser(
       email: email,
       testBrowser: _testBrowser,
       fakeEmailReader: _fakePubServerProcess.fakeEmailReader,
       scopes: scopes,
     );
   }
 }
+
+Future<TestUser> _createFakeTestUser({
+  required String email,
+  required TestBrowser testBrowser,
+  required FakeEmailReaderFromOutputDirectory fakeEmailReader,
+  List<String>? scopes,
+}) async {
+  late PubApiClient api;
+  await testBrowser.withPage(fn: (page) async {
+    await page.fakeAuthSignIn(email: email, scopes: scopes);
+    api = await _apiClientHttpHeadersFromSignedInSession(page);
+  });
```

Comment:
Maybe, we should do: `testBrowser.createSession` which returns a new incognito session.

Then we can use that session in `withBrowserPage` and in `api`, and we won't have to call `fakeAuthSignIn` more than once.. (I think an incognito mode session can have multiple pages/tags).

I think when we do actual integration tests we won't want to do the sign-in process more than once per user. It'll have non-trival overhead, as in it'll be slow.

Reply:
Good point. I'll try to do it in a follow-up PR.